### PR TITLE
Ui Merge

### PR DESCRIPTION
**Pull request by joncolon**
This includes bugfixes on the graphs in the stat page, image load fixes on the contactlist rv, and other incremental changes

### CODE REVIEW

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/Contact.java [hunk 1]
```diff
@@ -26,7 +26,6 @@ public long getTimeLastContacted() {
 
     public void setTimeLastContacted(long timeLastContacted) {
         this.timeLastContacted = System.currentTimeMillis();
-
     }
 
     public Contact() {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/Contact.java:31]: each constructor should call the last one with null parameters to prevent duplicate code

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/PicassoHelper.java [hunk 2]
```diff
@@ -13,11 +13,16 @@ public PicassoHelper(Context context) {
...
+    public void preloadImages(String string){
+        Picasso.with(context).load(string).fetch();
+    }
+
     public void loadImageFromUri(Uri uri, ImageView imageView){
-        Picasso.with(context).load(uri).into(imageView);
+        Picasso.with(context).load(uri).resize(1000, 1000).centerInside().into(imageView);
     }
 
     public void loadImageFromString(String uriString, ImageView imageView){
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/PicassoHelper.java:24]: call the other loadImageFromUri to prevent duplicate code

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 3]
```diff
@@ -32,46 +28,88 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_contact_list);
         Stetho.initializeWithDefaults(this);
 
+        PermissionChecker permissionChecker = new PermissionChecker(this, getApplicationContext());
+        permissionChecker.checkPermissions();
+
+        context = getApplicationContext();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:63]: move this up since you use Context earlier

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 4]
```diff
@@ -32,46 +28,88 @@
...
                 ContactListActivity.this.openEditor();
             }
         });
-
         setupRecyclerView();
         refreshRecyclerView();
-        checkPermissions();
-        buildInputContactDialog(this);
+        buildEnterContactDialog(this);
+//        checkServiceCreated();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:84]: remove

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 5]
```diff
@@ -32,46 +28,88 @@
...
     public static final String PARCELLED_CONTACT = "Parcelled Contact";
     private RecyclerView recyclerView;
+    private TextView importContactsTV;
     private AlertDialog InputContactDialogObject;
     private List<Contact> contactList;
     private SQLiteDatabase db;
-    private String mText = "";
-    private final String[] permissions = new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE,
-            Manifest.permission.READ_SMS, Manifest.permission.READ_CONTACTS};
+    private String name = "";
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:51]: Remove. You never use `name` before setting it so you don't need to store it.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 6]
```diff
@@ -32,46 +28,88 @@
...
 
 import static nl.qbusict.cupboard.CupboardFactory.cupboard;
 
 public class ContactListActivity extends AppCompatActivity implements AlertDialogCallback<String>,
-        ContactListAdapter.Listener, LoaderManager.LoaderCallbacks<Cursor> {
+        ContactListAdapter.Listener {
 
     public static final String PARCELLED_CONTACT = "Parcelled Contact";
     private RecyclerView recyclerView;
+    private TextView importContactsTV;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:47]: Remove. You never reference this outside of `onCreate()` so you don't need to store it.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 7]
```diff
@@ -32,46 +28,88 @@
...
     private AlertDialog InputContactDialogObject;
     private List<Contact> contactList;
     private SQLiteDatabase db;
-    private String mText = "";
-    private final String[] permissions = new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE,
-            Manifest.permission.READ_SMS, Manifest.permission.READ_CONTACTS};
+    private String name = "";
+    private Context context;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:52]: Remove. You never reference this outside of `onCreate()` so you don't need to store it.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 8]
```diff
@@ -102,25 +141,38 @@ public void onClick(DialogInterface dialog, int which) {
...
-    @Override
-    public void alertDialogCallback(String dialogFragmentText) {
-        mText = dialogFragmentText;
-        splitFirstandLastName();
+    private void sortContacts() {
+        List<Contact> contacts = contactList;
+        Collections.sort(contacts, new Comparator<Contact>() {
+            @Override
+            public int compare(Contact o1, Contact o2) {
+                return o1.getFirstName().compareToIgnoreCase(o2.getFirstName());
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:149]: What if they're the same?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 9]
```diff
@@ -102,25 +141,38 @@ public void onClick(DialogInterface dialog, int which) {
...
-            String firstName;
-            if (name.split("\\w+").length > 1) {
-                lastName = name.substring(name.lastIndexOf(" ") + 1);
-                firstName = name.substring(0, name.lastIndexOf(' '));
-            } else {
-                firstName = name;
+    private void preloadContactListImages() {
+        PicassoHelper pHelper = new PicassoHelper(context);
+        for (int i = 0; i < contactList.size(); i++) {
+            if (contactList.get(i).getBackgroundImage() != null) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:157]: Use local variables so you're not calling the same thing multiple times

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java [hunk 10]
```diff
@@ -129,86 +181,38 @@ private void splitFirstandLastName() {
...
-        adapter.setData(contactList);
-    }
-
     @Override
     public void onContactLongClicked(Contact contact) {
         cupboard().withDatabase(db).delete(contact);
         refreshRecyclerView();
     }
 
+//    public void checkServiceCreated() {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ContactListActivity.java:202]: Remove

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ImportContacts.java [hunk 11]
```diff
@@ -1,63 +1,35 @@
...
 import nyc.c4q.jonathancolon.inContaq.utlities.sqlite.ContactDatabaseHelper;
 
 import static android.provider.ContactsContract.CommonDataKinds.Email;
 import static android.provider.ContactsContract.CommonDataKinds.Phone;
 import static android.provider.ContactsContract.Contacts;
 import static nl.qbusict.cupboard.CupboardFactory.cupboard;
 
-public class MainActivity extends Activity {
+
+public class ImportContacts {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/ImportContacts.java:21]: my previous comments still apply to this class

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/SplashScreen.java [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+import nyc.c4q.jonathancolon.inContaq.R;
+
+
+public class SplashScreen extends AppCompatActivity {
+
+    private TextView inContaq;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/SplashScreen.java:14]: why do you need to store this as a field?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/SplashScreenActivity.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+import nyc.c4q.jonathancolon.inContaq.R;
+
+public class SplashScreenActivity extends AppCompatActivity {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/activities/SplashScreenActivity.java:11]: duplicate class?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/adapters/ContactFragmentPagerAdapter.java [hunk 14]
```diff
@@ -28,16 +28,18 @@ public int getCount() {
     public Fragment getItem(int pos) {
         switch (pos) {
             case 0:
-                return ContactInfoFragment.newInstance();
+                return new ContactInfoFragment();
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/adapters/ContactFragmentPagerAdapter.java:31]: remove newInstance() from each of these Fragments

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/adapters/PreCachingLayoutManager.java [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Created by jonathancolon on 3/13/17.
+ */
+
+public class PreCachingLayoutManager extends LinearLayoutManager {
+    private static final int DEFAULT_EXTRA_LAYOUT_SPACE = 1000;
+    private int extraLayoutSpace = -1;
+    private Context context;
+
+    public PreCachingLayoutManager(Context context) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/adapters/PreCachingLayoutManager.java:16]: each constructor should call the most complex constructor with null params to avoid duplicate code

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/fragments/ContactSmsFragment.java [hunk 16]
```diff
@@ -169,26 +152,23 @@ public void onClick(View v) {
         }
     }
 
-    private void enableClickListeners() {
-        contactImageIV.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
+    private void enableClickListeners(View view) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/fragments/ContactSmsFragment.java:155]: this no longer seems to attach OnClickListeners.. it seems to just do the logic of them??

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/fragments/MonthlyGraphFragment.java [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+import nyc.c4q.jonathancolon.inContaq.R;
+import nyc.c4q.jonathancolon.inContaq.contactlist.Contact;
+import nyc.c4q.jonathancolon.inContaq.contactlist.activities.ContactListActivity;
+import nyc.c4q.jonathancolon.inContaq.utlities.graphs.MonthlyGraph;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.SmsHelper;
+
+
+public class MonthlyGraphFragment extends Fragment {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/fragments/MonthlyGraphFragment.java:25]: DailyGraphFragment and MonthlyGraphFragment should subclass a general GraphFragment class to avoid duplicate code

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/graphs/MonthlyGraph.java [hunk 18]
```diff
@@ -188,6 +185,8 @@ synchronized private void loadGraph() {
         animateGraph();
     }
 
+    LineSet dataSetMonthly;
+    LineSet dataset;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/graphs/MonthlyGraph.java:189]: no need to make these fields, they don't seem to be used outside of `setGraphData()`

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyReceivedWorkerTask.java [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Override
+    protected TreeMap<Integer, Integer> doInBackground(DailyTaskParams... params) {
+        listSms = params[0].getdailyListSms();
+        dailyReceivedText = params[0].getDailyTexts();
+        return getSmsStats(listSms);
+    }
+
+    @Override
+    protected void onPostExecute(TreeMap<Integer, Integer> ret) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyReceivedWorkerTask.java:39]: if all you do is call the superclass method you don't need to override it.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyReceivedWorkerTask.java [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+        super.onPostExecute(ret);
+    }
+
+    private TreeMap<Integer, Integer> getSmsStats(ArrayList<Sms> list) {
+        ArrayList<String> receivedSms = new ArrayList<>();
+
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getTime().equals(Calendar.getInstance())) {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyReceivedWorkerTask.java:47]: extract `list.get(i).getTime()` to avoid duplicate work

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyReceivedWorkerTask.java [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+                receivedSms.add(list.get(i).getTime());
+            }
+        }
+        dailyReceivedText = getDailyTexts(receivedSms);
+        return dailyReceivedText;
+    }
+
+    private TreeMap<Integer, Integer> getDailyTexts(ArrayList<String> list) {
+        for (int i = 0; i < list.size(); i++) {
+            long lg = Long.parseLong(list.get(i));
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyReceivedWorkerTask.java:57]: `lg` isn't a great name. what is it?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailySentWorkerTask.java [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.ArrayList;
+import java.util.TreeMap;
+
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+
+/**
+ * Created by Hyun on 3/11/17.
+ */
+
+public class DailySentWorkerTask extends AsyncTask<DailyTaskParams, Void,TreeMap<Integer, Integer>> {
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailySentWorkerTask.java:17]: From the name I expected this to be pretty much the same as DailyReceivedWorkerTask, but it's not... why?

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyTaskParams.java [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+
+/**
+ * Created by Hyun on 3/11/17.
+ */
+
+public class DailyTaskParams {
+
+    private static ArrayList<Sms> dailyListSms;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyTaskParams.java:14]: shouldn't be static.

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyTaskParams.java [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+    private static TreeMap<Integer, Integer> dailyTexts;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/dailysms/DailyTaskParams.java:15]: shouldn't be static

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/DailyGraph.java [hunk 25]
```diff
@@ -0,0 +1,216 @@
...
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+
+import static android.graphics.Color.parseColor;
+import static com.db.chart.renderer.AxisRenderer.LabelPosition.NONE;
+
+/**
+ * Created by Hyun on 3/11/17.
+ */
+
+public class DailyGraph{
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/DailyGraph.java:31]: this seems to have a lot in common with MonthlyGraph. should also subclass a generic Graph class to hold shared logic

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/WordAverage.java [hunk 26]
```diff
@@ -0,0 +1,34 @@
+package nyc.c4q.jonathancolon.inContaq.utlities.sms;
+
+/**
+ * Created by jello on 3/11/17.
+ */
+
+public class WordAverage {
+
+    public static int countWords(String s){
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/WordAverage.java:9]: what is this meant to do? I think there's an easier way...

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sqlite/ContactDatabaseHelper.java [hunk 27]
```diff
@@ -13,6 +13,8 @@
     private static final String DATABASE_NAME = "contact.db";
     private static final int DATABASE_VERSION = 1;
     private static ContactDatabaseHelper instance;
+    private Context context;
```
Comment (robertjli) [inline app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sqlite/ContactDatabaseHelper.java:16]: why did you add these two? you don't seem to use them?

**Review by robertjli — CHANGES_REQUESTED** [review]

**Comment by joncolon** [comment]
Merging so I can submit an apk. Will address these comments before next merge

File: app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/PicassoHelper.java (hunk 2 above)
Comment (joncolon) [reply app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/PicassoHelper.java:24]: I'm confused. loadImageFromUri is when you first grab a photo from the gallery, loadImageFromString is when loading from the database.

Comment (robertjli) [reply app/src/main/java/nyc/c4q/jonathancolon/inContaq/contactlist/PicassoHelper.java:24]: They do the exact same thing, except loadImageFromString() just converts the String to a Uri using Uri.parse(uriString). So you could just do

```java
loadImageFromString(String uriString, ImageView imageView) {
    loadImageFromUri(Uri.parse(uriString), imageView);
}
```
